fem/pnonlinearform.cpp

```diff
@@ -65,6 +65,39 @@ void ParNonlinearForm::Mult(const Vector &x, Vector &y) const
 
    NonlinearForm::Mult(X, Y);
 
+   if (fbfi.Size())
+   {
+      // Terms over shared interior faces in parallel.
+      ParFiniteElementSpace *pfes = ParFESpace();
+      ParMesh *pmesh = pfes->GetParMesh();
+      FaceElementTransformations *tr;
+      const FiniteElement *fe1, *fe2;
+      Array<int> vdofs1, vdofs2;
+      Vector el_x, el_y;
+
+      X.ExchangeFaceNbrData();
+      for (int i = 0; i < pmesh->GetNSharedFaces(); i++)
+      {
+         tr = pmesh->GetSharedFaceTransformations(i, true);
+
+         fe1 = pfes->GetFE(tr->Elem1No);
+         fe2 = pfes->GetFaceNbrFE(tr->Elem2No);
+
+         pfes->GetElementVDofs(tr->Elem1No, vdofs1);
+         pfes->GetFaceNbrElementVDofs(tr->Elem2No, vdofs2);
+
+         el_x.SetSize(vdofs1.Size() + vdofs2.Size());
+         X.GetSubVector(vdofs1, el_x.GetData());
+         X.FaceNbrData().GetSubVector(vdofs2, el_x.GetData() + vdofs1.Size());
```

Comment:
Sorry, not intended for a full review, but I don't understand this line

el_x.GetData() + vdofs1.Size()

They seem to be two different datatypes. What is this supposed to do.

Reply:
Thanks for looking this over! The intention is for `el_x` to store both DOFs for the self element (`vdofs1`), as well as the face-neighbor element (`vdofs2`) to pass to `AssembleFaceVector`. Line 90 fills the self element data, then 91 adds the face-neighbor data after the self element data.